include/benchmark/benchmark.h

```diff
@@ -1244,7 +1244,7 @@ template <class Lambda, class... Args>
 internal::Benchmark* RegisterBenchmark(const char* name, Lambda&& fn,
                                        Args&&... args) {
   return benchmark::RegisterBenchmark(
-      name, [=](benchmark::State& st) { fn(st, args...); });
+      name, [=, gn=std::forward<Lambda>(fn)](benchmark::State& st) { gn(st, args...); });
```

Comment:
CI is really not happy about doing this unconditionally.
I guess you need to add a BENCHMARK_HAS_CXX14 macro (see how BENCHMARK_HAS_CXX11 is defined),
and do
```
#if defined(BENCHMARK_HAS_CXX14)
return benchmark::RegisterBenchmark(
      name, [=, gn=std::forward<Lambda>(fn)](benchmark::State& st) { gn(st, args...); });
#else
  return benchmark::RegisterBenchmark(
      name, [=](benchmark::State& st) { fn(st, args...); });
#endif
```
(+clang-format)